students/terrance_jones/lesson5/mailroom_part3.py

```diff
@@ -0,0 +1,169 @@
+#list of donors
+donor_list = {'Carlos Santos':[25,50,100], 'Esperanza Gomez': [10,20,30], 'Paul Jackson':[5,10,15], 'Karl Black':[100,200,300], 'Charles Exx': [15,30,45]}
+
+
+"""
+FUNCTIONS NEEDED FOR SCRIPT TO RUN
+"""
+def get_name():
+    """get donor name from user"""
+    name = input("Enter the full name of the donor.")
+    return name
+
+def show_donor_list():
+    """Create list of donor names only. Display list of all donors"""
+    for key in donor_list.keys():
+        print(key)
+    
+
+def find_donor(x):
+    for key in donor_list.keys():
+        if key == x:
+            return True
```

Comment:
no need for a loop; dictionaries are very efficient with key look ups -> `if key in donor_list` will have the same effect.